### PR TITLE
Issue form template

### PR DESCRIPTION
**Pull request by ramyaparimi**
### Why:

Issue forms for creating new issues in OS repo
* [Improve existing docs](https://github.com/github/docs/blob/issue-form-template/.github/ISSUE_TEMPLATE/improve-existing-docs.yaml)
* [Improve the site](https://github.com/github/docs/blob/issue-form-template/.github/ISSUE_TEMPLATE/improve-the-site.yml)

### What's being changed:

Converted .md files to .yml issue forms 

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,7 @@ coverage/
 /data/early-access
 dist
 .next
+package-lock.json
```
Comment (heiskr) [inline .gitignore:14]: We'll need to revert this change, as we do need package-lock for our actions processes and for security reasons.

Comment (heiskr) [reply .gitignore]: 👍🏼 If you can also run `git checkout main package-lock.json`

Comment (ramyaparimi) [reply .gitignore]: Ran the command as well!!

Comment (heiskr) [reply .gitignore]: Thanks!

**Comment by heiskr** [comment]
`npm ci; npm run prettier` should get your build green

**Review by heiskr — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: Improve existing content
+description: Make a suggestion to improve the content in an existing article
+title: "['Improve existing content']: "
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:3]: This looks like we'll be encouraging people to have titles like `['Improve existing content']: What I really want the title to be`. Personally, I'd prefer the title to be `What I really want the title to be` because it makes issues more searchable. For example, if we have, over time, hundreds of issues with "improve existing content" in the title, then we won't be able to meaningfully search for anything using the words "improve", "exist", or "content".

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+labels:
+  - content
+body:
+  # HUBBERS BEWARE! THE GITHUB/DOCS REPO IS PUBLIC TO THE ENTIRE INTERNET. OPEN AN ISSUE IN GITHUB/DOCS-CONTENT INSTEAD.
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:7]: Should this be a part of the Markdown section below? I feel like since nobody will be looking at the YAML file except those of us maintaining it, we should either include it in the Markdown section or simply remove it.

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        For questions, ask in [Discussions](https://github.com/github/docs/discussions)
+        Before you file an issue read the
+        - [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)
+        - [Contributing guide](https://github.com/github/docs/blob/main/CONTRIBUTING.md)
+
+  - type: checkboxes
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:16]: I feel like an "I agree to comply with the Code of Conduct" checkbox is going to be more useful here than a "I have searched the existing issues" checkbox. There's no cost to the issue filer for checking and not complying and no benefit to us saying, "But you said you'd do it!". Whereas the Code of Conduct checkbox is an agreement to be bound by certain rules that include penalties for not following the rules.

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    attributes:
+      label: What article on docs.github.com is affected?
+      placeholder: |
+        Please link to the article you'd like to see updated
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: What part(s) of the article would you like to see updated?
+      placeholder: |
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:34]: It looks like multi-line content doesn't do well as a `placeholder`. Maybe this should be `description` instead?

File: .gitignore [hunk 6]
```diff
@@ -12,6 +12,7 @@ coverage/
 dist
 .next
 
+
```
Comment (lee-dohm) [inline .gitignore:15]: You should probably revert this change since it seems unintentional 😀

**Comment by lee-dohm** [comment]
If you're converting from Markdown issue templates to YAML issue templates, this PR should also delete the Markdown versions.

**Comment by ramyaparimi** [comment]
@lee-dohm Thanks for the review and suggestions! I made the changes and also created an issue form for [Partner Program](https://github.com/github/docs/blob/issue-form-template/.github/ISSUE_TEMPLATE/partner-contributed-documentation.yml). Could you please give another look at them? 🙇‍♀️

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml (hunk 3 above)
Comment (ramyaparimi) [reply .github/ISSUE_TEMPLATE/improve-existing-docs.yaml]: I added as a Markdown. Also changed the wording a little, let me know if it looks alright!

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+        * For questions, ask in [Discussions](https://github.com/github/docs/discussions)
+        * Before you file an issue read the [Contributing guide](https://github.com/github/docs/blob/main/CONTRIBUTING.md)
+        * Check to make sure someone hasn't already opened a similar [issue](https://github.com/github/docs/issues)
+
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Code of Conduct
+      description: By submitting this issue, you agree to follow our [Code of Conduct](https://example.com)
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:20]:
```suggestion
      description: This project has a [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md) that all participants are expected to understand and follow.
```

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      options:
+        - label: I agree to follow this project's Code of Conduct
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:22]:
```suggestion
        - label: I have read and agree to the GitHub Docs project's Code of Conduct.
```

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+          required: true
+
+  - type: textarea
+    attributes:
+      label: What article on docs.github.com is affected?
+      description: |
+        Please link to the article you'd like to see updated
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:29]:
```suggestion
      description: Please link to the article you'd like to see updated.
```

Since this is a single line, you don't really need the pipe.

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        - Why should the docs be changed? What use cases does it support? 
+        - What is the expected outcome?
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Additional information
+      description: |
+        Add any other context or screenshots about the feature request here.
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:47]:
```suggestion
      description: Add any other context or screenshots about the feature request here.
```

Comment (ramyaparimi) [reply .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:46]: Got it!

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 11]
```diff
@@ -0,0 +1,49 @@
+name: Improve existing content
+description: Make a suggestion to improve the content in an existing article
+title: '[What I really want the title to be]'
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:3]: I don't think that `title` is a required field, is it? Can we just leave this out?

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+labels:
+  - content
+body:
+  - type: markdown
+    attributes:
+      value: |
+        **HUBBERS!! THIS IS GITHUB/DOCS OPEN SOURCE REPO. OPEN AN ISSUE IN GITHUB/DOCS-CONTENT INSTEAD.**
```
Comment (lee-dohm) [inline .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:10]:
```suggestion
        **HUBBERS:** This is the github/docs open source repo. You may want to open an issue in the internal-only github/docs-content repo instead.
```

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml (hunk 11 above)
Comment (ramyaparimi) [reply .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:3]: Yea I agree! I had the same question while writing the other two forms as well. Can we leave it out in the other two forms as well?

Comment (lee-dohm) [reply .github/ISSUE_TEMPLATE/improve-existing-docs.yaml:3]: Yes.

**Comment by lee-dohm** [comment]
I have some significant questions about the intent of the Partner Program issue template. Can you remove it from this PR and open a separate PR for it?

**Comment by ramyaparimi** [comment]
> I have some significant questions about the intent of the Partner Program issue template. Can you remove it from this PR and open a separate PR for it?

Alright! Will do

File: .github/ISSUE_TEMPLATE/improve-existing-docs.yaml (hunk 11 above)
Comment (ramyaparimi) [reply .github/ISSUE_TEMPLATE/improve-existing-docs.yaml]: Deleted the `title` key

**Comment by ramyaparimi** [comment]
> I have some significant questions about the intent of the Partner Program issue template. Can you remove it from this PR and open a separate PR for it?

Here is the PR https://github.com/github/docs/pull/8618

**Review by jaystyles — DISMISSED** [review]

**Comment by ramyaparimi** [comment]
@lee-dohm Is this now ready to ship? 😄

**Review by lee-dohm — DISMISSED** [review]
Looks great!

One final change you should make before merging is removing the deletion of the `partner-contributed-documentation.md` file from this PR 😀

**Review by lee-dohm — DISMISSED** [review]
:shipit:

**Comment by ramyaparimi** [comment]
@heiskr thank you for helping me with getting the build green ✨ 
@lee-dohm Thanks so much for taking time to review and give feedback on this PR 💜 Merging it now

**Comment by ramyaparimi** [comment]
@lee-dohm Thank you for helping me with resolving the issues in this PR 💖  It was a ton of helpful knowledge 😀  ✨

**Review by lee-dohm — APPROVED** [review]
Looks good to me 👍
